source/agora/consensus/protocol/Nominator.d

```diff
@@ -223,15 +246,24 @@ extern(D):
 
         Signs the SCPEnvelope with the node's private key.
 
-        todo: Currently not signing yet. To be done.
-
         Params:
             envelope = the SCPEnvelope to sign
 
     ***************************************************************************/
 
     public override void signEnvelope (ref SCPEnvelope envelope)
     {
+        scope (failure) assert(0);
+        import core.stdc.stdlib;
+
+        auto msg = toHash(envelope.statement)[];
+        auto sig = this.key_pair.secret.sign(msg)[];
+
+        // note: SCP free's the memory, it must be malloc-allocated.
+        // see also: #547
+        auto mem = cast(ubyte*)malloc(sig.length);
```

Comment:
We need to avoid malloc calls. A signature (as the way we define it) is fixed-size, so there shouldn't be a need to allocate memory.

We should change `SCPEnvelope` to embed a fixed-length signature in place of the vector which it currently uses.

File an issue for it and fix it before moving on with this PR.